### PR TITLE
Update Contributing Guide

### LINKED ISSUE #6111: Update CONTRIBUTING.md on bugfixes/features PRs
## 📚 Documentation
I might've missed the context, but we recently changed the strategy of which branches feature/bugfix PRs should get merged into, didn't we? If so, the docs should reflect it.

https://github.com/PyTorchLightning/pytorch-lightning/blob/4f63942c4d89d3417eefc9b98884738ee6ff015e/.github/CONTRIBUTING.md#L333

### PR DESCRIPTION
**Pull request by kaushikb11**
## What does this PR do?

Master is the 👑 now.
Fixes #6111

## Before submitting
- [x] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [ ] Did you make sure to update the documentation with your changes? (if necessary)
- [ ] Did you write any new necessary tests? (not for typos and docs)
- [ ] Did you verify new and existing tests pass locally with your changes?
- [ ] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [ ] Check that all items from **Before submitting** are resolved
 - [ ] Make sure the title is self-explanatory and the description concisely explains the PR
 - [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: .github/CONTRIBUTING.md [hunk 1]
```diff
@@ -323,50 +323,6 @@ run our/your test with
...
-#### How to contribute bugfixes/features?
-
-Currently we have separate streams/branches for bugfixes/features and release from the default branch (`master`).
-Bugfixes should land in this `master` branch and features should land in `release/X.y-dev`.
-This means that when starting your contribution and creating a branch according to question 2) you should start this new branch from master or future release dev branch.
-Later in PR creation also pay attention to properly set the target branch, usually the starting (base) and target branch are the same.
-
-_Note, that this flow may change after the 1.2 release as we will adjust releasing strategy._
-
-#### How to fix PR with mixed base and target branches?
```
Comment (Borda) [inline .github/CONTRIBUTING.md:335]: would keep this section, as we still may have some PR from past branch which shall be now master

**Review by Borda — CHANGES_REQUESTED** [review]

File: .github/CONTRIBUTING.md (hunk 1 above)
Comment (kaushikb11) [reply .github/CONTRIBUTING.md:335]: @Borda But we could remove the `How to contribute bugfixes/features` section, right?

Comment (Borda) [reply .github/CONTRIBUTING.md:335]: yes, agree :]

Comment (kaushikb11) [reply .github/CONTRIBUTING.md:335]: @Borda Done

**Review by Borda — APPROVED** [review]

**Review by SeanNaren — APPROVED** [review]

**Review by carmocca — APPROVED** [review]

**Review by s-rog — APPROVED** [review]

**Pull request merged**